Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabLoader.cpp

```diff
@@ -620,9 +620,8 @@ namespace AzToolsFramework
 
             // Now that the Instance has been created, convert it back into a Prefab DOM.  Because we
             // don't specify to skip default fields in the call to StoreInstanceInPrefabDom,
-            // this new DOM will have all fields from all classes except link ids.
-            if (!PrefabDomUtils::StoreInstanceInPrefabDom(
-                    loadedPrefabInstance, loadedTemplateDomRef, PrefabDomUtils::StoreFlags::StripLinkIds))
+            // this new DOM will have all fields from all classes.
+            if (!PrefabDomUtils::StoreInstanceInPrefabDom(loadedPrefabInstance, loadedTemplateDomRef))
```

Comment:
Because it was removing link ids from nested template doms upon the initial load. That was causing some issues during testing where we expect the linkId to be present but its not. So, I changed.

I cannot think of any reasons why we would want the initial load to remove linkIds at all. Might have been an oversight.

Reply:
Discussed offline that this flag should be removed, otherwise link ids in nested instance DOM (e.g. Car) that is stored in the template DOM (e.g. Level) would be removed. But we also need to add new logic to remove link id in the top-level template DOM.

Reply:
Yes. In fact, this is the reason 'PrefabLoadTemplateTests' fail in the AR run. Once I remove linkId from the sanitized template dom, the tests pass now.